src/App.tsx

```diff
@@ -16,46 +16,88 @@ export const goodsFromServer = [
 ];
 
 export const App: React.FC = () => {
+  const goods = [...goodsFromServer];
+  const [sortType, setSortType] = useState('none');
+  const [isReversed, setReverse] = useState(false);
+
+  const sortByAlphabet = () => {
+    setSortType('alphabet');
+  };
+
+  const sortByLength = () => {
+    setSortType('length');
+  };
+
+  const changeReversState = () => {
+    setReverse(!isReversed);
+  };
+
+  const reseting = () => {
+    setSortType('none');
+    setReverse(false);
+  };
+
+  switch (sortType) {
+    case 'alphabet':
+      goods.sort((g1, g2) => g1.localeCompare(g2));
+      break;
+
+    case 'length':
+      goods.sort((g1, g2) => g1.length - g2.length);
+      break;
+
+    default:
+      break;
+  }
+
+  if (isReversed) {
+    goods.reverse();
+  }
```

Comment:
create separate function for sort goods. 
So that it works like that `const visibleGoods = sortGoods(goods)`
[Optional] use `useMemo` here